tests/test_update_picking.py

```diff
@@ -0,0 +1,503 @@
+from . import common
+from odoo.exceptions import ValidationError
+
+class TestGoodsInUpdatePicking(common.BaseUDES):
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestGoodsInUpdatePicking, cls).setUpClass()
+        # Pretty much TestGoodsInPicking Setup
+        User = cls.env['res.users']
+        PickingType = cls.env['stock.picking.type']
+        Picking = cls.env['stock.picking']
+        user_warehouse = User.get_user_warehouse()
+        # Get goods in type
+        in_type_id = user_warehouse.in_type_id
+        cls.picking_type_in = PickingType.search([('id', '=', in_type_id.id)])
+        # Setting default source location as goods_in doesn't have one
+        cls.picking_type_in.default_location_src_id = cls.env.ref('stock.stock_location_suppliers')
+
+    def test01_update_picking_validate_complete(self):
+        """ Test update_picking by addding by splitting a product
+            without serial numbers
+        """
+        create_info = [{'product': self.apple, 'qty': 4}]
+        picking = self.create_picking(self.picking_type_in,
+                                      origin='test_picking_origin',
+                                      products_info=create_info,
+                                      confirm=True)
+
+        products_info = [{'product_barcode': self.apple.barcode, 'qty': 4}]
+        picking.update_picking(products_info=products_info)
```

Comment:
what about doing the following: find the move line before update_picking, assert that qty_done = 0, call update_picking, assert that the operation is now 4.